src/controllers/v1/_common/demand.ts

```diff
@@ -216,9 +267,29 @@ export class DemandController extends Controller {
     const [demand] = await this.db.get('demand', { id: demandId, subtype: this.dbDemandSubtype })
     if (!demand) throw new NotFound(this.demandType)
 
+    if (!(await this.canAccessDemand(demand))) {
+      throw new Unauthorized('You are not allowed to view this demand')
+    }
+
     const dbTxs = await this.db.get('transaction', query)
     return dbTxs.map(dbTransactionToResponse)
   }
+
+  public async canAccessDemand(demand: DemandRow, readOnly: boolean = true): Promise<boolean> {
```

Comment:
I'm going to suggest instead of using `readOnly` as a boolean we pass an `action` variable which is something like `"read" | "createOnChain"`. That will make it easier to extend. Also don't use a default here make it explicit